File: Problems & Solutions/2389. Longest Subsequence With Limited Sum/25 December 2022/Approach 1 - Sort and Count/Longest Subsequence With Limited Sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        # Sort 'nums'
        nums.sort()
        ans = []

        for query in queries:
            count = 0
            for num in nums:
                if query >= num:
                    query -= num
                    count += 1
                else:
                    break
            ans.append(count)
        
        return ans
        

File: Problems & Solutions/2389. Longest Subsequence With Limited Sum/25 December 2022/Approach 1 - Sort and Count/Longest Subsequence With Limited Sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& nums, vector<int>& queries) {
        sort(nums.begin(), nums.end());
        
        vector<int> ans;
        for (auto query : queries) {
            int count = 0;
            for (auto num : nums) {
                if (query >= num) {
                    query -= num;
                    count++;
                }
                else {
                    break;
                }
            }
            ans.push_back(count);
        }
        
        return ans;
    }
};